processor/src/main/java/org/mapstruct/ap/processor/MethodRetrievalProcessor.java

```diff
@@ -80,19 +93,60 @@ public int getPriority() {
         return 1;
     }
 
+    private List<SourceMethod> retrievePrototypeMethods(TypeMirror typeMirror, MapperConfig mapperConfig) {
+        if ( typeMirror == null || typeMirror.getKind() == TypeKind.VOID ) {
+            return Collections.emptyList();
+        }
+
+        TypeElement typeElement = asTypeElement( typeMirror );
+        List<SourceMethod> methods = new ArrayList<SourceMethod>();
+        for ( ExecutableElement executable : getAllEnclosedExecutableElements( elementUtils, typeElement ) ) {
+
+            ExecutableType methodType = typeFactory.getMethodType( typeElement, executable );
+            List<Parameter> parameters = typeFactory.getParameters( methodType, executable );
+            boolean containsTargetTypeParameter = SourceMethod.containsTargetTypeParameter( parameters );
+
+            // prototype methods don't have prototypes themselves
```

Comment:
> The only way to have inheritance relation between return types in mapping method is as the base return type is in a mapping method in the MappingConfig class AFAIK.

Why would that be? `@InheritConfiguration` can refer to methods declared on the same mapper (that was its purpose prior to this PR). I'm confused, maybe we are talking about different things?

> But, perhaps the similar reason as you state above. To render a direct mapping invisible to the user. The generated forged method could then be private.

Yes, ok. Although it's seems a very special scenario. Why would one expose an iterable mapping method but not the element mapping method? I don't mind if we do not support this, at least for now. There is nothing wrong with it per se, only I think it'll hardly be used in practice.